### ISSUE TITLE
alabaster.matrix.check failure

### ISSUE THREAD

**Issue by martin-g** [Yikun/BBS#62]
```
biocbuild@kunpeng1 ~/git> R CMD build --keep-empty-dirs --no-resave-data alabaster.matrix
* checking for file ‘alabaster.matrix/DESCRIPTION’ ... OK
* preparing ‘alabaster.matrix’:
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘alabaster.matrix_0.99.1.tar.gz’
```

```
biocbuild@kunpeng1 ~/git> R CMD check alabaster.matrix_0.99.1.tar.gz 
* using log directory ‘/home/biocbuild/git/alabaster.matrix.Rcheck’
* using R Under development (unstable) (2023-03-12 r83975)
* using platform: aarch64-unknown-linux-gnu (64-bit)
* R was compiled by
    gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
    GNU Fortran (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
* running under: Ubuntu 22.04.2 LTS
* using session charset: UTF-8
* checking for file ‘alabaster.matrix/DESCRIPTION’ ... OK
* this is package ‘alabaster.matrix’ version ‘0.99.1’
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘alabaster.matrix’ can be installed ... OK
* checking installed package size ... OK
* checking package directory ... OK
* checking ‘build’ directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking installed files from ‘inst/doc’ ... OK
* checking files in ‘vignettes’ ... OK
* checking examples ... ERROR
Running examples in ‘alabaster.matrix-Ex.R’ failed
The error most likely occurred in:

> ### Name: writeSparseMatrix
> ### Title: Write a sparse matrix
> ### Aliases: writeSparseMatrix
> 
> ### ** Examples
> 
> library(Matrix)
> x <- rsparsematrix(100, 20, 0.5)
> tmp <- tempfile(fileext=".h5")
> writeSparseMatrix(x, tmp, "csc_matrix")
The 'cset' argument has been deprecated.
Please use the argument 'encoding' instead.
The 'cset' argument has been deprecated.
Please use the argument 'encoding' instead.
> writeSparseMatrix(x, tmp, "csr_matrix", column=FALSE)
The 'cset' argument has been deprecated.
Please use the argument 'encoding' instead.
The 'cset' argument has been deprecated.
Please use the argument 'encoding' instead.
Error in H5Dset_extent(did, dims) : HDF5. Dataset. Can't set value.
Calls: writeSparseMatrix ... .blockwise_sparse_writer -> h5set_extent -> H5Dset_extent
Execution halted
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
    1. ├─alabaster.base::stageObject(x, dir, file.path(experiment, assay)) at test-stage-matrix.R:142:4
    2. └─alabaster.matrix::stageObject(x, dir, file.path(experiment, assay))
    3.   └─alabaster.matrix (local) .local(x, dir, path, child, ...)
    4.     └─alabaster.matrix:::.stage_delayed(...)
    5.       └─alabaster.matrix (local) fallback(x, dir, path, child = child)
    6.         └─alabaster.matrix:::.stage_sparse_matrix(x, dir, path, child = child)
    7.           └─alabaster.matrix::writeSparseMatrix(...)
    8.             └─alabaster.matrix:::.write_CS_matrix(...)
    9.               └─alabaster.matrix:::.blockwise_sparse_writer(...)
   10.                 └─rhdf5::h5set_extent(file, iname, newlast)
   11.                   └─rhdf5::H5Dset_extent(did, dims)
  
  [ FAIL 6 | WARN 0 | SKIP 0 | PASS 49 ]
  Error: Test failures
  Execution halted
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in ‘inst/doc’ ... OK
* checking running R code from vignettes ...
  ‘userguide.Rmd’ using ‘UTF-8’... OK
 NONE
* checking re-building of vignette outputs ... OK
* checking PDF version of manual ... OK
* DONE

Status: 2 ERRORs
See
  ‘/home/biocbuild/git/alabaster.matrix.Rcheck/00check.log’
for details.
```

**Comment by martin-g** [comment]
Reported upstream - https://github.com/ArtifactDB/alabaster.matrix/issues/2

**Comment by martin-g** [comment]
This issue has been fixed upstream!

**Comment by martin-g** [comment]
Also https://github.com/grimbough/rhdf5/issues/123 is fixed in 2.43.3

**Comment by martin-g** [comment]
https://bioconductor.org/checkResults/3.18/bioc-LATEST/alabaster.matrix/kunpeng2-checksrc.html is fully green!
